src/ripple/rpc/handlers/WalletPropose.cpp

```diff
@@ -31,9 +31,30 @@
 #include <ripple/rpc/handlers/WalletPropose.h>
 #include <ed25519-donna/ed25519.h>
 #include <boost/optional.hpp>
+#include <cmath>
+#include <map>
 
 namespace ripple {
 
+double
+shannon_entropy (std::string const& input)
+{
+    std::map<int, double> freq;
+
+    for (auto const& c : input)
+        freq[c]++;
+
+    double se = 0;
+
+    for (auto& f : freq)
+        f.second /= input.length();
+
+    for (auto const& f : freq)
```

Comment:
Ultra minor thought.  I might be inclined to move the declaration of `se` down to just in front of the second `for` loop -- just in front of its first usage.